### PR TITLE
Use std::unordered_map for TiffTreeStruct lookup

### PR DESCRIPTION
**Pull request by kevinbackhouse**
Same idea as #2299: use `std::unordered_map` for faster lookup.

### CODE REVIEW

File: src/tiffimage_int.hpp [hunk 1]
```diff
@@ -155,17 +155,16 @@ using TiffGroupTable = std::unordered_map<TiffGroupKey, NewTiffCompFct, TiffGrou
          use standard TIFF layout.
 */
 struct TiffTreeStruct {
-  using Key = std::pair<uint32_t, IfdId>;
-  //! Comparison operator to compare a TiffTreeStruct with a TiffTreeStruct::Key
-  bool operator==(const Key& key) const;
+  const IfdId parentGroup_;      //!< Parent group
```
Comment (neheb) [inline src/tiffimage_int.hpp:158]: why are these const?

Comment (kevinbackhouse) [reply src/tiffimage_int.hpp]: You're right: I didn't need to do that. I also didn't need to add the constructor. I've removed them.

File: src/tiffimage_int.hpp [hunk 2]
```diff
@@ -155,12 +155,8 @@ using TiffGroupTable = std::unordered_map<TiffGroupKey, NewTiffCompFct, TiffGrou
          use standard TIFF layout.
 */
 struct TiffTreeStruct {
```
Comment (neheb) [inline src/tiffimage_int.hpp:157]: Now this is a struct with two variables. Why not just use std::pair?

**Comment by neheb** [comment]
:crossed_swords: Conflicts

File: src/tiffimage_int.cpp [hunk 3]
```diff
@@ -1830,8 +1830,8 @@ void TiffCreator::getPath(TiffPath& tiffPath, uint32_t extendedTag, IfdId group,
     tiffPath.push(TiffPathItem(extendedTag, group));
```
Comment (neheb) [inline src/tiffimage_int.cpp:1830]: Oh this is funny. clang-tidy should but is not recommending emplace() here.

Comment (kevinbackhouse) [reply src/tiffimage_int.cpp]: fixed

**Review by neheb — APPROVED** [review]

**Pull request merged**